### DISCUSSION TITLE
Recommend best VPS for self hosting SL?

### DISCUSSION THREAD

**Discussion by whisperdancer** [simple-login/app#1373] [Self-hosting question]
I've been able to get SL setup properly. The top issue I'm experiencing is allowing outgoing port 25 and IP reputation for email. I've been reading through these discussions and it seems as though Amazon VPS is not a good choice for VPS for SL due to restricting port 25 and then even after you send customer support request to allow outgoing on port 25, the IP may work for a while, but at any time you run the risk of your IP being blacklisted. 

As such, is there a recommended list of VPS firms that allow outgoing port 25 and have a good IP reputation that would be ideal for self hosting SL? Please understand I"m seeking recommendations, not endorsements. Perhaps others in the self host community can chime in who are having a good experience with their VPS of choice. I believe this thread will be very helpful for those looking to self host successfully. 

Thank you.

**Comment by DANgerous25** [comment]
Hi, I originally went with Dreamhost but had issues because the IP was somehow blacklisted. I spoke to DH and they recommended not using them for VPS solutions where email was concerned as it was too easy for the IP range to be blacklisted and very difficult to get off it. In the end, I’m self-hosting my instance and using DH as a mail relay to get around my ISP blocking output port 25.

I know that this isn’t the direct answer you’re looking for but just sharing my experience after going down a similar path to you.

**Reply by whisperdancer** [reply]
@DANgerous25 Thanks for sharing. Curious, when self hosting do you have to open port 25 and 443 on your router? And if so, how to you mitigate the potential security risk of having these ports open, especially port 25.

Thank you

**Comment by nguyenkims** [comment]
@whisperdancer at SL, we used to use UpCloud and are quite happy with their service. We needed to go through a small process in order to unblock the port 25 but this is a good thing as this avoids spammers from abusing the VPS easily which leads to bad IP reputation.

**Reply by whisperdancer** [reply]
@nguyenkims Thank you for sharing. I haven't heard of UpCloud and they have a SJ data center so this might be a good choice.

**Comment by mzch** [comment]
@whisperdancer I run SL on [LetBox!](https://letbox.com/) with incoming/outgoing via port 25. I also tried SL on [Linode](https://www.linode.com/) without problem.

**Reply by whisperdancer** [reply]
@mzch Thanks for sharing. Curious, with Linode or Letbox was port 25 closed by default and you had to contact their support team to open port 25 for you?

**Reply by mzch** [reply]
? They leave port 25 open by default now, so I installed Potfix and SL only. if needed firewall, you can configure in CP.

**Reply by whisperdancer** [reply]
@mzch Thank you. Great to know port 25 is open by default. However, as @nguyenkims  stated above having to request the port open helps avoid spammers using the VPS. If abused by spammers, it runs the risk of IP being blacklisted. Unfortunately Letbox is out of stock. How long did you use Linode and did the IP ever get blacklisted? Also, Firewall configure in CP. What is CP? Thank you!

**Comment by DNCD** [comment]
I run SL on NFOservers without any issues. You will need to open a ticket to get port 25 open. It's blocked by default.
